File: nomad-samples/python/movies/content_groups/content_groups.py
from content_group.add_content_to_content_group import *
from content_group.create_content_group import *
from content_group.delete_content_group import*
from content_group.get_content_group import *
from content_group.get_content_groups import *
from content_group.remove_content_from_content_group import *
from content_group.rename_content_group import *
from content_group.share_content_group_with_user import *
from content_group.stop_sharing_content_group_with_user import *
from login.login import *

def login_main():
    while True:
        USERNAME = "xxxxxxxx"#input("Enter username: ")
        PASSWORD = "***REMOVED***"#input("Enter password: ")
        AUTH_TOKEN = login(USERNAME, PASSWORD)["token"]
        if not AUTH_TOKEN:
            print("Invalid credentials")
        else:
            return AUTH_TOKEN

def get_content_group_main(AUTH_TOKEN):
    try:
        CONTENT_GROUP_ID = input("Input an content group id: ")

        print("Getting content groups")
        INFO = get_content_group(AUTH_TOKEN, CONTENT_GROUP_ID)
        print(json.dumps(INFO, indent=4))
    except:
        raise Exception()
    

def get_content_groups_main(AUTH_TOKEN):
    try:
        print("Getting content groups")
        INFO = get_content_groups(AUTH_TOKEN)
        print(json.dumps(INFO, indent=4))
    except:
        raise Exception()

def create_content_groups_main(AUTH_TOKEN):
    try:
        CONTENT_GROUP_NAME = input("Input an content group name (press enter to skip): ")

        print("Creating a new content group") 
        INFO = create_content_group(AUTH_TOKEN, CONTENT_GROUP_NAME)
        print(json.dumps(INFO, indent=4))
    except:
        raise Exception()

def add_content_to_content_group_main(AUTH_TOKEN):
    try:
        CONTENT_GROUP_ID = input("Input an content group id: ")
        CONTENTS = input("Enter content ids you wish to add to an content group (separate by comma): ").replace(" ","")

        print("Adding content to content group")
        INFO = add_content_to_content_group(AUTH_TOKEN, CONTENT_GROUP_ID, CONTENTS.split(","))
        print(json.dumps(INFO, indent=4))
    except:
        raise Exception() 

def remove_content_from_content_group_main(AUTH_TOKEN):
    try:
        CONTENT_GROUP_ID = input("Input an content group id: ")
        CONTENTS = input("Enter content ids you wish to remove from an content group (separate by comma): ").replace(" ","")

        print("Removing content from content group")
        INFO = remove_content_to_content_group(AUTH_TOKEN, CONTENT_GROUP_ID, CONTENTS.split(","))
        print(json.dumps(INFO, indent=4))
    except:
        raise Exception()

def rename_content_group_main(AUTH_TOKEN):
    try: 
        CONTENT_GROUP_ID = input("Input an content group id: ")
        CONTENT_GROUP_NEW_NAME = input("Input a new name for an content group: ")

        print("Renaming content group")
        INFO = rename_content_group(AUTH_TOKEN, CONTENT_GROUP_ID, CONTENT_GROUP_NEW_NAME)
        print(json.dumps(INFO, indent=4))
    except:
        raise Exception()

def share_content_with_user_main(AUTH_TOKEN):
    try:
        CONTENT_GROUP_ID = input("Input an content group id: ")
        USER_ID_ARR = []
        while True:
            USER_ID = input("Enter the user id of the user you wish to share the content with: ")
            USER_ID_ARR.append(USER_ID)

            INPUT = input("Do you want to add another user to the list? (y/n): ")
            if INPUT == "n":
                break 

        print("Sharing content with user")
        INFO = share_content_group_with_user(AUTH_TOKEN, CONTENT_GROUP_ID, USER_ID_ARR)
        print(json.dumps(INFO, indent=4))
    except:
        raise Exception()

def stop_sharing_content_with_user_main(AUTH_TOKEN):
    try:
        CONTENT_GROUP_ID = input("Input an content group id: ")
        USER_ID_ARR = []
        while True:
            USER_ID = input("Enter the user id of the user you wish to stop sharing the content with: ")
            USER_ID_ARR.append(USER_ID)

            INPUT = input("Do you want to add another user to the list? (y/n): ")
            if INPUT == "n":
                break 

        print("Sharing content with user")
        INFO = stop_sharing_content_group_with_user(AUTH_TOKEN, CONTENT_GROUP_ID, USER_ID_ARR)
        print(json.dumps(INFO, indent=4))
    except:
        raise Exception()
    
def delete_content_group_main(AUTH_TOKEN):
    try:
        CONTENT_GROUP_ID = input("Input an content group id: ")

        print("Deleting content group")
        INFO = delete_content_group(AUTH_TOKEN, CONTENT_GROUP_ID)
        print(json.dumps(INFO, indent=4))
    except:
        raise Exception()

if __name__ == "__main__":
    AUTH_TOKEN = login_main()
    
    while True:
        print("Do you want to get a specific content group, get your content group, create an "\
              "content group, add content to a content group, remove content from a content group, "\
              "rename a content, , share content with a user, stop sharing content with a user, "\
              "delete content or exit")
        USER_INPUT = input("Enter get groups, get group, create, add, remove, rename, start "\
                           "sharing, stop sharing, get portal, delete, or exit for each option "\
                           "above respectively: ")

        if USER_INPUT == "get group":
            get_content_group_main(AUTH_TOKEN)
        elif USER_INPUT == "get groups":
            get_content_groups_main(AUTH_TOKEN)
        elif USER_INPUT == "create":
            create_content_groups_main(AUTH_TOKEN)
        elif USER_INPUT == "add":
            add_content_to_content_group_main(AUTH_TOKEN)
        elif USER_INPUT == "remove":
            remove_content_from_content_group_main(AUTH_TOKEN)
        elif USER_INPUT == "rename":
            rename_content_group_main(AUTH_TOKEN)
        elif USER_INPUT == "start sharing":
            share_content_with_user_main(AUTH_TOKEN)
        elif USER_INPUT == "stop sharing":
            stop_sharing_content_with_user_main(AUTH_TOKEN)
        elif USER_INPUT == "delete":
            delete_content_group_main(AUTH_TOKEN)
        elif USER_INPUT == "exit":
            break
        else:
            print("Input invalid")

File: nomad-samples/portal/js/content-groups/content-groups.js
import addContentToContentGroup from "./content-groups/add-content-to-content-group.js";
import createContentGroup from "./content-groups/create-content-group.js";
import deleteContentGroup from "./content-groups/delete-content-group.js";
import getContentGroup from "./content-groups/get-content-group.js";
import getContentGroups from "./content-groups/get-content-groups.js";
import getPortalGroups from "./content-groups/get-potal-groups.js";
import removeContentFromContentGroup from "./content-groups/remove-contents-from-content-group.js";
import renameContentGroup from "./content-groups/rename-content-group.js";

const AUTH_FORM = document.getElementById("authForm");
const GET_GROUP_FORM = document.getElementById("getGroupForm");
const GET_GROUPS_FORM = document.getElementById("getGroupsForm");
const CREATE_FORM = document.getElementById("createForm");
const ADD_FORM = document.getElementById("addForm");
const REMOVE_FORM = document.getElementById("removeForm");
const RENAME_FORM = document.getElementById("renameForm");
const PORTAL_FORM = document.getElementById("portalForm");
const DELETE_FORM = document.getElementById("deleteForm");

const TOKEN_INPUT = document.getElementById("authInput");
const GET_GROUP_ID_INPUT = document.getElementById("getGroupIdInput");
const NAME_INPUT = document.getElementById("nameInput");
const ADD_INPUT = document.getElementById("addInput");
const ADD_CONTENT_INPUT = document.getElementById("addContentInput");
const REMOVE_INPUT = document.getElementById("removeInput");
const REMOVE_CONTENT_INPUT = document.getElementById("removeContentInput");
const RENAME_GROUP_ID_INPUT = document.getElementById("renameInput");
const RENAME_GROUP_INPUT = document.getElementById("renameGroupInput");
const PORTAL_GROUP_INPUT = document.getElementById("portalGroupInput");
const DELETE_INPUT = document.getElementById("deleteInput");

sessionStorage.clear();

AUTH_FORM.addEventListener("submit", function (event)
{
    event.preventDefault();
    sessionStorage.setItem("token", TOKEN_INPUT.value);
    console.log("Successfuly updated token");
});

GET_GROUP_FORM.addEventListener("submit", function (event)
{
    event.preventDefault();

    let id = GET_GROUP_ID_INPUT.value;

    getContentGroupMain(id);
});

GET_GROUPS_FORM.addEventListener("submit", function (event)
{
    event.preventDefault();
    getContentGroupsMain();
});

CREATE_FORM.addEventListener("submit", function (event)
{
    event.preventDefault();

    let name = NAME_INPUT.value;

    createContentGroupMain(name);
});

ADD_FORM.addEventListener("submit", function (event)
{
    event.preventDefault();

    let contentGroup = ADD_INPUT.value;
    let contents = ADD_CONTENT_INPUT.value;

    addContentToContentGroupMain(contentGroup, contents);
});

REMOVE_FORM.addEventListener("submit", function (event)
{
    event.preventDefault();

    let contentGroup = REMOVE_INPUT.value;
    let contents = REMOVE_CONTENT_INPUT.value

    removeContentFromContentGroupMain(contentGroup, contents);
});

RENAME_FORM.addEventListener("submit", function (event)
{
    event.preventDefault();

    let contentGroupId = RENAME_GROUP_ID_INPUT.value;
    let renameGroup = RENAME_GROUP_INPUT.value;

    renameContentGroupMain(contentGroupId, renameGroup);
});

PORTAL_FORM.addEventListener("submit", function (event)
{
    event.preventDefault();

    let portalGroups = PORTAL_GROUP_INPUT.value;

    getPortalGroupsMain(portalGroups);
});

DELETE_FORM.addEventListener("submit", function (event)
{
    event.preventDefault();

    let contentGroup = DELETE_INPUT.value

    deleteContentGroupMain(contentGroup);
});

async function getContentGroupsMain()
{
    if (!sessionStorage.getItem("token"))
    {
        throw new Error("Authentication token: The authentication token is empty");
    }

    try
    {
        console.log("Getting content groups");
        const CONTENT_GROUPS = await getContentGroups(sessionStorage.getItem("token"));
        console.log(JSON.stringify(CONTENT_GROUPS, null, 4));
    }
    catch (error)
    {
        throw new Error(error);
    }
}

async function getContentGroupMain(ID)
{
    if (!sessionStorage.getItem("token"))
    {
        throw new Error("Authentication token: The authentication token is empty");
    }

    try
    {
        console.log("Getting content group");
        const INFO = await getContentGroup(sessionStorage.getItem("token"), ID)
        console.log(JSON.stringify(INFO, null, 4));
    }
    catch (error)
    {
        throw new Error(error);
    }
}

async function createContentGroupMain(NAME)
{
    if (!sessionStorage.getItem("token"))
    {
        throw new Error("Authentication token: The authentication token is empty");
    }

    try
    {
        console.log("Creating content groups");
        const CONTENT_GROUPS = await createContentGroup(sessionStorage.getItem("token"), NAME);
        console.log(JSON.stringify(CONTENT_GROUPS, null, 4));
    }
    catch (error)
    {
        throw new Error(error);
    }
}

async function addContentToContentGroupMain(CONTENT_GROUP, CONTENTS)
{
    if (!sessionStorage.getItem("token"))
    {
        throw new Error("Authentication token: The authentication token is empty");
    }

    try
    {
        console.log("Adding contents to content group");
        const CONTENT_GROUPS = await addContentToContentGroup(sessionStorage.getItem("token"), CONTENT_GROUP, CONTENTS.split(","));
        console.log(JSON.stringify(CONTENT_GROUPS, null, 4));
    }
    catch (error)
    {
        throw new Error(error);
    }
}

async function removeContentFromContentGroupMain(CONTENT_GROUP, CONTENTS)
{
    if (!sessionStorage.getItem("token"))
    {
        throw new Error("Authentication token: The authentication token is empty");
    }

    try
    {
        console.log("Removeing content groups");
        const CONTENT_GROUPS = await removeContentFromContentGroup(sessionStorage.getItem("token"), CONTENT_GROUP, CONTENTS.split(","));
        console.log(JSON.stringify(CONTENT_GROUPS, null, 4));
    }
    catch (error)
    {
        throw new Error(error);
    }
}

async function renameContentGroupMain(CONTENT_GROUP, NAME)
{
    if (!sessionStorage.getItem("token"))
    {
        throw new Error("Authentication token: The authentication token is empty");
    }

    try
    {
        console.log("Renaming content groups");
        const CONTENT_GROUPS = await renameContentGroup(sessionStorage.getItem("token"), CONTENT_GROUP, NAME);
        console.log(JSON.stringify(CONTENT_GROUPS, null, 4));
    }
    catch (error)
    {
        throw new Error(error);
    }
}

async function getPortalGroupsMain(PORTAL_GROUPS)
{
    if (!sessionStorage.getItem("token"))
    {
        throw new Error("Authentication token: The authentication token is empty");
    }

    try
    {
        console.log("Retrieving portal groups");
        const INFO = await getPortalGroups(sessionStorage.getItem("token"), PORTAL_GROUPS.split(","));
        console.log(JSON.stringify(INFO, null, 4));
    }
    catch (error)
    {
        throw new Error(error);
    }
}

async function deleteContentGroupMain(CONTENT_GROUP)
{
    if (!sessionStorage.getItem("token"))
    {
        throw new Error("Authentication token: The authentication token is empty");
    }

    try
    {
        console.log("Deleting content groups");
        const CONTENT_GROUPS = await deleteContentGroup(sessionStorage.getItem("token"), CONTENT_GROUP);
        console.log(JSON.stringify(CONTENT_GROUPS, null, 4));
    }
    catch (error)
    {
        throw new Error(error);
    }
}

